packages/perf/lib/index.js

```diff
@@ -47,12 +56,35 @@ class FirebasePerfModule extends FirebaseModule {
   constructor(...args) {
     super(...args);
     this._isPerformanceCollectionEnabled = this.native.isPerformanceCollectionEnabled;
+    this._instrumentationEnabled = true;
   }
 
   get isPerformanceCollectionEnabled() {
     return this._isPerformanceCollectionEnabled;
   }
 
+  get instrumentationEnabled() {
+    return this._instrumentationEnabled;
+  }
+
+  set instrumentationEnabled(enabled) {
+    if (!isBoolean(enabled)) {
+      throw new Error("firebase.perf().instrumentationEnabled = 'enabled' must be a boolean.");
+    }
+    this._instrumentationEnabled = enabled;
+  }
+
+  get dataCollectionEnabled() {
+    return this.isPerformanceCollectionEnabled;
+  }
+
+  set dataCollectionEnabled(enabled) {
+    if (!isBoolean(enabled)) {
+      throw new Error("firebase.perf().dataCollectionEnabled = 'enabled' must be a boolean.");
+    }
+    this._isPerformanceCollectionEnabled = enabled;
```

Comment:
I think right here should be the current implementation of `setPerformanceCollectionEnabled` (code just below this comment) so that it actually turns on or off the native logging of perf events. Right now this doesn't actually do anything, but `this.native.setPerformanceCollectionEnabled(enabled)` is the effect it *should* have I think?

Then setPerformanceCollectionEnabled should delegate to this in it's implementation, and in the future state where it is gone completely this method (with the native call...) is all that's left + and it works

Reply:
I think you're right. It won't be await'ed (i.e. it's not a promise) but I guess it doesn't matter because it only takes effect the next time the app is run 🤔

Reply:
If I understand you correctly, I don't think `setPerformanceCollectionEnabled` can delegate to this as it is a setter. You cannot return a setter.

Reply:
You cannot return a setter, but maybe you can call it internally? Just thinking how to de-duplicate the code (DRY principle) but that's an implementation detail. As long as the backwards-compatible method and the setter are actually *doing* the necessary thing, it's okay

Reply:
I think the best we can do is:

```js
 set dataCollectionEnabled(enabled) {
    if (!isBoolean(enabled)) {
      throw new Error("firebase.perf().dataCollectionEnabled = 'enabled' must be a boolean.");
    }
    this.setPerformanceCollectionEnabled(enabled);
  }
```

No way of de-duplicating the Error I'm afraid, but it will at least throw the correct error message for the correct API.

Reply:
That works for me!